### ISSUE TITLE
Potential memory leak?

### ISSUE THREAD

**Issue by Some-random** [samsungsds-rnd/deepspeech.mxnet#8]
As I'm training the model (1000 hours of data, batch size is 100 on 4 GPUs, code stays pretty much unchanged), I noticed the CPU memory usage gets higher and higher and eventually the system has to kill the process. 

So I'm wondering whether others have had this problem too, and if so, could there be a potential memory leak in the code?

**Comment by Soonhwan-Kwon** [comment]
For my case, as I remember, batch_size of 12, cpu memory stays at 32gb when both is_batchnorm and is_bucketing are True and when is_batchnorm=True and is_bucketing=False, cpu memory stays at 26gb. And I trained model for a month, I guess there is no leak as far as I know. Would you tell us how much memory do you have? And maybe batchsize of 100 for 4 GPUs are too large.(It's just a guess) Did your MXNet goes down after it loaded the gpu memory or before?

**Comment by Some-random** [comment]
I have 256gb of memory. My MXNet goes down a fews days after I started my model because the system won't care about the memory usage until it's getting too high.

I guess the only difference between my model and yours is I'm not using batchnorm (I'm also using bucketing). Also, the increase of CPU memory usage is pretty recognizable once you start training.

**Comment by Soonhwan-Kwon** [comment]
Would you provide mxnet log/cfg file/what's your version of MXNetcurrently using and what's is the value of /proc/sys/vm/vfs_cache_pressure ? I didn't experienced memory leak until MXNet 0.10.1, if you provides above information, it may help me to rule out some possibilities and thank you for reporting the issue!

**Comment by Soonhwan-Kwon** [comment]
Because you don't give any information like cfg file, I'll close this issue. 
But there is a possible cause that if you set buckets too many, it can cause OOM. I recommend you to set small number of buckets.
